timemachine/cpp/src/nonbonded_interaction_group.cu

```diff
@@ -351,6 +341,82 @@ void NonbondedInteractionGroup<RealType>::execute_device(
     }
 }
 
+template <typename RealType>
+void NonbondedInteractionGroup<RealType>::verify_row_indices(const std::vector<int> &row_indices) {
+    if (row_indices.size() == 0) {
+        throw std::runtime_error("idxs can't be empty");
+    }
+    std::set<int> unique_idxs(row_indices.begin(), row_indices.end());
+    if (unique_idxs.size() != row_indices.size()) {
+        throw std::runtime_error("atom indices must be unique");
+    }
+    if (*std::max_element(row_indices.begin(), row_indices.end()) >= N_) {
+        throw std::runtime_error("indices values must be less than N");
+    }
+    if (*std::min_element(row_indices.begin(), row_indices.end()) < 0) {
+        throw std::runtime_error("indices values must be greater or equal to zero");
+    }
+}
```

Comment:
nit: DRY with `NonbondedAllPairs::verify_atom_idxs`?

Reply:
Resolved in [bcadefe](https://github.com/proteneer/timemachine/pull/970/commits/bcadefe01cd7941eee5d6d9cbb9c98eec053aa3e)

Found a bug where you could segfault if you set nonbonded atom idxs to 0. It was also inconsistent with how I had modified the NBlist for setting indices, so now it constructs a full NBlist rather than one of size K. Actually going to change that back after thinking a bit more. Will just throw a NBlist error now.